apps/calculation/calculation.cpp

```diff
@@ -66,20 +66,44 @@ KDCoordinate Calculation::height(Context * context, bool expanded) {
     Layout inputLayout = createInputLayout();
     KDCoordinate inputHeight = inputLayout.layoutSize().height();
+    KDCoordinate inputWidth = inputLayout.layoutSize().width();
+    float singleMargin = 2 * Metric::CommonSmallMargin;
+    float doubleMargin = 4 * Metric::CommonSmallMargin;
+    bool singleLine = false;
     if (display == DisplayOutput::ExactOnly) {
       KDCoordinate exactOutputHeight = createExactOutputLayout().layoutSize().height();
-      *memoizedHeight = inputHeight+exactOutputHeight;
+      KDCoordinate exactOutputWidth = createExactOutputLayout().layoutSize().width();
+      singleLine = exactOutputWidth + inputWidth < maxWidth - 40;
+      if (singleLine) {
+        *memoizedHeight = (inputHeight >= exactOutputHeight) ? inputHeight + singleMargin : exactOutputHeight + singleMargin;
+      } else {
+        *memoizedHeight = inputHeight + exactOutputHeight + doubleMargin;
+      }
     } else if (display == DisplayOutput::ApproximateOnly || (!expanded && display == DisplayOutput::ExactAndApproximateToggle)) {
       KDCoordinate approximateOutputHeight = createApproximateOutputLayout(context).layoutSize().height();
-      *memoizedHeight = inputHeight+approximateOutputHeight;
+      KDCoordinate approximateOutputWidth = createApproximateOutputLayout(context).layoutSize().width();
+      singleLine = approximateOutputWidth + inputWidth < maxWidth - 40;
+      if (singleLine) {
+        *memoizedHeight = (inputHeight >= approximateOutputHeight) ? inputHeight + singleMargin : approximateOutputHeight + singleMargin;
+      } else {
+        *memoizedHeight = inputHeight + approximateOutputHeight + doubleMargin;
+      }
     } else {
       assert(display == DisplayOutput::ExactAndApproximate || (display == DisplayOutput::ExactAndApproximateToggle && expanded));
       Layout approximateLayout = createApproximateOutputLayout(context);
       Layout exactLayout = createExactOutputLayout();
       KDCoordinate approximateOutputHeight = approximateLayout.layoutSize().height();
       KDCoordinate exactOutputHeight = exactLayout.layoutSize().height();
       KDCoordinate outputHeight = maxCoordinate(exactLayout.baseline(), approximateLayout.baseline()) + maxCoordinate(exactOutputHeight-exactLayout.baseline(), approximateOutputHeight-approximateLayout.baseline());
-      *memoizedHeight = inputHeight + outputHeight;
+      KDCoordinate exactOutputWidth = exactLayout.layoutSize().width();
+      KDCoordinate approximateOutputWidth = approximateLayout.layoutSize().width();
+      KDCoordinate outputWidth = exactOutputWidth + approximateOutputWidth;
+      singleLine = outputWidth + inputWidth < maxWidth - 70;
+      if (singleLine) {
+        *memoizedHeight  = (inputHeight >= outputHeight) ? inputHeight + singleMargin : outputHeight + singleMargin;
+      } else {
+        *memoizedHeight  = inputHeight + outputHeight + doubleMargin;
+      }
     }
     /* For all display output except ExactAndApproximateToggle, the selected
```

Comment:
```suggestion
     if (expanded) {
         *memoizedHeight = *memoizedHeight + singleMargin;
     }
     /* For all display output except ExactAndApproximateToggle, the selected
```

Reply:
I compiled it with this change, but I'm not sure what bug it is fixing.

Reply:
There is a bug with the expanded view where it either doesn’t display at all or it displays incorrectly. Try entering something like `0.8/0.3` and then pressing the up arrow after executing. After the change, it should correctly display the fractional answer along side the decimal answer when selected.